src/transformers/models/llama/modeling_llama.py

```diff
@@ -999,39 +999,94 @@ def forward(
         all_self_attns = () if output_attentions else None
         next_decoder_cache = None
 
-        for decoder_layer in self.layers:
+        ###################################################################
```

Comment:
Can you offload that method in a new utility method `_create_gradient_checkpointing_custom_forward`? and we use the output of that method to pass it directly into `self._gradient_checkpointing_func`

Reply:
but since each model class has different input arguments, i don't think there is a universal method to cover all model classes.